src/Mollie/WC/Payment/Payment.php

```diff
@@ -593,4 +485,50 @@ public function refund( WC_Order $order, $order_id, $payment_object, $amount = n
 		}
 	}
 
+    /**
+     * @param WC_Order $order
+     * @param WC_Payment_Gateway $gateway
+     * @param                    $newOrderStatus
+     * @param                    $orderId
+     */
+    protected function maybeUpdateStatus(
+        WC_Order $order,
+        WC_Payment_Gateway $gateway,
+        $newOrderStatus,
+        $orderId
+    ) {
+        if (!$this->isOrderPaymentStartedByOtherGateway($order)) {
+            if ($gateway || ($gateway instanceof Mollie_WC_Gateway_Abstract)) {
+                $gateway->updateOrderStatus($order, $newOrderStatus);
+            }
+        } else {
+            $orderPaymentMethodTitle = get_post_meta(
+                $orderId,
+                '_payment_method_title',
+                $single = true
+            );
+
+            // Add message to log
+            Mollie_WC_Plugin::debug(
+                $gateway->id . ': Order ' . $order->get_id()
+                . ' webhook called, but payment also started via '
+                . $orderPaymentMethodTitle . ', so order status not updated.',
+                true
+            );
+
+            // Add order note
+            $order->add_order_note(
+                sprintf(
+                /* translators: Placeholder 1: payment method title, placeholder 2: payment ID */
+                    __(
+                        'Mollie webhook called, but payment also started via %s, so the order status is not updated.',
+                        'mollie-payments-for-woocommerce'
+                    ),
+                    $orderPaymentMethodTitle
+                )
+            );
+        }
+    }
```

Comment:
You have a very similar function in `Order` class. Is this is what you want to have? If so, probably it makes sense to create a trait for it. Or just make it public and call from outside, if this makes sense in this case.